accounts-db/src/append_vec.rs

```diff
@@ -348,9 +376,26 @@ impl AppendVec {
 
     /// when we can use file i/o as opposed to mmap, this is the trigger to tell us
     /// that no more appending will occur and we can close the initial mmap.
+    #[allow(dead_code)]
```

Comment:
```
        #[cfg(not(unix))]
        // must open as mmmap on non-unix
        return None;
```

because of this not unix, we get clippy errors on non-unix. Thus, the dead_code